frontend/src/scenes/insights/insightVizDataLogic.ts

```diff
@@ -444,6 +451,7 @@ const handleQuerySourceUpdateSideEffects = (
     currentState: InsightQueryNode
 ): QuerySourceUpdate => {
     const mergedUpdate = { ...update } as InsightQueryNode
+    const trendsMergedUpdate = mergedUpdate as TrendsQuery
```

Comment:
This is pretty dirty and the intention is hard to read. I only have an idea what's this for, because I see in the removed diff how `;(mergedUpdate as Partial<TrendsQuery>)` was annoying. But I think we should leave that as it was in this PR